kernel/src/process/process/mod.rs

```diff
@@ -771,7 +770,11 @@ impl Process {
     fn propagate_has_child_subreaper(&self) {
         let mut process_queue = VecDeque::new();
         let children = self.children().lock();
-        for child_process in children.values() {
+        let Some(children_ref) = children.as_ref() else {
+            // The current process is exiting group at the same time.
```

Comment:
The current process should not be a zombie process. This is unnecessary.

Anyway, this isn't a big problem, so I'm merging it. As I mentioned earlier, I'm going to address several other issues that were identified during the review of this PR, so in fact, I'll need to touch the code again.